src/httpc.c

```diff
@@ -464,6 +464,36 @@ httpc_set_accept_encoding(struct httpc_request *req, const char *encoding)
 #endif
 }
 
+int
+httpc_set_http_version(struct httpc_request *req, const char *version)
+{
+	if (strcmp(version, "1.0") == 0){
+		curl_easy_setopt(req->curl_request.easy, 
+			CURL_HTTP_VERSION_1_0, version);
+	}
+	if (strcmp(version, "1.1") == 0){
+                curl_easy_setopt(req->curl_request.easy,
+                        CURL_HTTP_VERSION_1_1, version);
+        }
+	if (strcmp(version, "2") == 0){
+                curl_easy_setopt(req->curl_request.easy,
+                        CURL_HTTP_VERSION_2, version);
+        }
+	if (strcmp(version, "2-tls") == 0){
+                curl_easy_setopt(req->curl_request.easy,
+                        CURL_HTTP_VERSION_2TLS, version);
+        }
+	if (strcmp(version, "2-prior-knowledge") == 0){
+                curl_easy_setopt(req->curl_request.easy,
+                        CURL_HTTP_VERSION_2_PRIOR_KNOWLEDGE, version);
+        }
+	else{
+		diag_set(IllegalParams, "Not a supported http version");
```

Comment:
I would add the value into the error. It helps wth debugging.

Example:

https://github.com/tarantool/tarantool/blob/655c144e7e0a556fe9a8f45c2a9e900d351623cb/src/box/lua/key_def.c#L183-L184

Reply:
Fixed with "Not a supported http version: %s", version